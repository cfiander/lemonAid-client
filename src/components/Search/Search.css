.ingredientsList {
    font-family: 'Playfair Display', serif;
  font-size: 300%;
  text-align: center;
  font-weight: bold;
}

.addIngredientButton {
  color:white;
}