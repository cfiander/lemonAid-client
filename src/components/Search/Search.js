import React, { Component } from 'react'
import RecipeContext from '../../contexts/RecipeContext'
import uuid from 'uuid'
import logo from '../../images/Lemon.jpg'
import ValidationError from '../../ValidationError'

import { } from "@fortawesome/free-solid-svg-icons";
import { } from "@fortawesome/react-fontawesome";

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      formValid: false,
      validationMessages: {
        name: '',
      }
    }
  }

  static contextType = RecipeContext;

  static defaultProps = {
    addIngredient: () => { },
  }

  updateName(name) {
    this.setState({ name }, () => { this.validateName(name) });
  }

  validateName(fieldValue) {
    console.log(this.state)
    const fieldErrors = { ...this.state.validationMessages };
    let hasError = false;

    fieldValue = fieldValue.trim();

    if (fieldValue.length < 3) {
      fieldErrors.name = 'Ingredients must be at least 3 characters long';
      hasError = true;
    } else {
      fieldErrors.name = '';
      hasError = false;
    }

    this.setState({
      validationMessages: fieldErrors,
      nameValid: !hasError
    }, this.formValid);
  }

  handleSubmit = (e) => {
    e.preventDefault();
    if (!this.state.nameValid) {
      return;
    }
    const ingredient = {
      key: uuid(),
      stateId: uuid(),
      name: this.state.name
    }
    this.context.addIngredient(ingredient)
  }

  render() {
    return (
      <>
        <h1 className="ingredientTitle">Ingredient Table</h1>
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="name"
            id="ingredientName"
            placeholder="Type ingredient"
            ref={input => this.search = input}
            onChange={e => this.updateName(e.target.value)}
          />
          <button
            type="submit"
            className="addIngredientButton">
            <img className="logo" alt="logo" src={`${logo}`} />
            <div className="plus">&#x2b;</div>
          </button>
          <ValidationError hasError={!this.state.nameValid} message={this.state.validationMessages.name}/>  
        </form>
      </>
    )
  }
}

export default Search
