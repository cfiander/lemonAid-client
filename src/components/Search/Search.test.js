import React from 'react'
import { shallow } from 'enzyme'
import Search from './Search';
import toJson from 'enzyme-to-json'

describe('Search Form',() =>{
    it('renders a search by default', () => {
        const wrapper = shallow(<Search />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})