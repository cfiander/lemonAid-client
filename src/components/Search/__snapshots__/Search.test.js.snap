// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders a search by default 1`] = `
<Fragment>
  <h1
    className="ingredientTitle"
  >
    Ingredient Table
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      id="ingredientName"
      name="name"
      onChange={[Function]}
      placeholder="Add an ingredient..."
      type="text"
    />
    <button
      className="addIngredientButton"
      type="submit"
    >
      <img
        alt="logo"
        className="logo"
        src="Lemon.jpg"
      />
      <div
        className="plus"
      >
        +
      </div>
    </button>
  </form>
</Fragment>
`;
